openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -101,10 +102,24 @@ def dispatch(self, request, *args, **kwargs):
         response = super(AccessTokenView, self).dispatch(request, *args, **kwargs)
 
         if response.status_code == 200 and request.POST.get('token_type', '').lower() == 'jwt':
-            expires_in, scopes, user = self._decompose_access_token_response(request, response)
+            client_id = self._get_client_id(request)
+            adapter = self.get_adapter(request)
+            expires_in, scopes, user = self._decompose_access_token_response(adapter, response)
+            issuer, secret, audience, filters = self._get_client_specific_claims(client_id, adapter)
 
             content = {
-                'access_token': JwtBuilder(user).build_token(scopes, expires_in),
+                'access_token': JwtBuilder(
+                    user,
+                    secret=secret,
+                    issuer=issuer,
+                ).build_token(
+                    scopes,
+                    expires_in,
+                    aud=audience,
+                    additional_claims={
+                        'filters': filters,
```

Comment:
Here, `filters` are put in the `additional_claims` section of the token.  However, when [parsing the token](https://github.com/edx/edx-drf-extensions/blob/f1e51a082e090dd88c3543e45e4727e0c55106ae/edx_rest_framework_extensions/utils.py#L122), the filters are coming from the top-level field in the token.  Or did I mis-read this?

Reply:
Yes, it's a bit tricky but `decode_jwt_filters` calls `jwt_decode_handler` which returns the inner payload or `access_token` part of the token. We are good here.